WikiFetcher.java

```diff
@@ -0,0 +1,83 @@
+package com.flatironschool.javacs;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+
+public class WikiFetcher {
+	private long lastRequestTime = -1;
+	private long minInterval = 1000;
+
+	/**
+	 * Fetches and parses a URL string, returning a list of paragraph elements.
+	 *
+	 * @param url
+	 * @return
+	 * @throws IOException
+	 */
+	public Elements fetchWikipedia(String url) throws IOException {
+		sleepIfNeeded();
+
+		// download and parse the document
+		Connection conn = Jsoup.connect(url);
+		Document doc = conn.get();
+
+		// select the content text and pull out the paragraphs.
+		Element content = doc.getElementById("mw-content-text");
+
+		// TODO: avoid selecting paragraphs from sidebars and boxouts
+		Elements paras = content.select("p");
+		return paras;
+	}
+
+	/**
+	 * Reads the contents of a Wikipedia page from src/resources.
+	 *
+	 * @param url
+	 * @return
+	 * @throws IOException
+	 */
+	public Elements readWikipedia(String url) throws IOException {
+		URL realURL = new URL(url);
+
+		// assemble the file name
+		String slash = File.separator;
+		String filename = "resources" + slash + realURL.getHost() + realURL.getPath();
+
+		// read the file
+		InputStream stream = WikiFetcher.class.getClassLoader().getResourceAsStream(filename);
+		Document doc = Jsoup.parse(stream, "UTF-8", filename);
+
```

Comment:
Once you have your Document object, the code of this method is thereafter identical to fetchWikipedia(). You should factor out that common code into a single method.